### ISSUE TITLE
Allow backticks as valid "other" in quotemark rule "avoid-escape" option

### ISSUE THREAD

**Issue by bportnoy** [palantir/tslint#3306]
### Bug Report

TSLint version: 5.7.0
TypeScript version: 2.4.2
Running TSLint via: (pick one) CLI

#### TypeScript code being linted

```ts
console.log(`we're loving this rule`)
```

with `tslint.json` configuration:

```json
      "quotemark": [true, "single", "jsx-double", "avoid-escape", "avoid-template"]
```

#### Actual behavior
Error generated:
```  
Error  (quotemark) Lint
    ` should be "
```

#### Expected behavior
Strings denoted by backticks should be accepted as a valid if they allow the avoidance of escaping.

**Comment by ajafff** [comment]
I guess that makes sense. But only if `"avoid-template"` is not present.

**Comment by thomasward1212** [comment]
Has anyone started work on this?

**Comment by ajafff** [comment]
@thomasward1212 If you want to tackle this issue, go for it. It doesn't seem anyone has already started work.

**Comment by thomasward1212** [comment]
@ajafff Awesome, thanks.

**Comment by charles-salmon** [comment]
@ajafff I don't believe this is actually an issue. In the example provided in the Bug Report, the rule configuration includes `avoid-template`, and thus the behaviour is as expected. Removal of this option also removes the error.

**Comment by cheeZery** [comment]
I agree with @charles-salmon, this issue should be closed. @bportnoy a template string doesn't become one by using placeholders, the backticks itself are enough ;-) So with the option "avoid-template" the linter works just as expected.

**Comment by JoshuaKGoldberg** [comment]
Sounds like this can be closed from the discussion, and it's been open for a while. Closing. 

If you think it should be re-opened, please do comment!
